packages/api/src/useInfiniteQuery.ts

```diff
@@ -0,0 +1,42 @@
+import { useInfiniteQuery as _useInfiniteQuery } from '@tanstack/react-query';
+import useAPI from './useAPI';
+import type {
+    TSocketAcceptableProps,
+    TSocketPaginateableEndpointNames,
+    TSocketRequestInfiniteQueryOptions,
+    TSocketRequestPayload,
+} from '../types';
+import { getQueryKeys } from './utils';
+
+const useInfiniteQuery = <T extends TSocketPaginateableEndpointNames>(
+    name: T,
+    ...props: TSocketAcceptableProps<T, true, 'useInfiniteQuery'>
+) => {
+    const prop = props?.[0];
+    const payload = prop && 'payload' in prop ? (prop.payload as TSocketRequestPayload<T>) : undefined;
+    const options = prop && 'options' in prop ? (prop.options as TSocketRequestInfiniteQueryOptions<T>) : undefined;
+    const { send } = useAPI();
+
+    // @ts-expect-error The `offset` parameter is always present in the `payload` for the paginateable endpoints.
+    const initial_offset = payload?.offset || 0;
```

Comment:
Can we create a default payload if it doesn't exist so that we can avoid the lint ignores?